forge-game/src/main/java/forge/game/ability/effects/ConniveEffect.java

```diff
@@ -79,15 +79,15 @@ public void resolve(SpellAbility sa) {
 
                 p.drawCards(num, sa, moveParams);
 
-                CardCollectionView validCards = p.getCardsIn(ZoneType.Hand);
-                validCards = CardLists.filter(validCards, CardPredicates.Presets.NON_TOKEN);
-                if (validCards.isEmpty() || !p.canDiscardBy(sa, true)) { // hand being empty unlikely, just to be safe
+                CardCollection validDisards = (CardCollection) p.getCardsIn(ZoneType.Hand);
+                validDisards = CardLists.filter(validDisards, CardPredicates.Presets.NON_TOKEN);
```

Comment:
so you don't need to cast:

`CardLists.filter(p.getCardsIn(ZoneType.Hand), CardPredicates.Presets.NON_TOKEN)`